### PR TITLE
Add Qwen3 MoE model

### PR DESCRIPTION
**Pull request by pcmoritz**
This PR adds the full Qwen3 MoE model definition. Since the MoE model is almost the same as the fully connected model, I am just re-using most of the code between the two.

The current setting is good for fine-tuning, but not for pre-training (there is no expert balancing loss for example).

### CODE REVIEW

**Comment by pcmoritz** [comment]
Testes the PR with:
```
uv run --with huggingface_hub hf download DavidAU/Qwen3-MOE-4x0.6B-2.4B-Writing-Thunder-V1.2 --local-dir /tmp/qwen3_moe

uv run --extra gpu --with wandb --with jinja2 tx train --model DavidAU/Qwen3-MOE-4x0.6B-2.4B-Writing-Thunder-V1.2 --dataset HuggingFaceH4/ultrachat_200k --split train_sft --output-dir /tmp --batch-size 4 --loader tx.loaders.chat --load-checkpoint-path /tmp/qwen3_moe --tp-size 8 --tracker wandb --tracker-args '{"name": "Qwen3-MOE-4x0.6B-2.4B", "project": "tx"}'

uv run --with huggingface_hub hf download DavidAU/Qwen3-MOE-4x0.6B-2.4B-Writing-Thunder-V1.2 --local-dir /tmp/ultrachat

cp /tmp/model.safetensors /tmp/ultrachat/

uv run --with vllm vllm serve /tmp/ultrachat/

curl http://localhost:8000/v1/completions -H "Content-Type: application/json" -d '{
        "model": "/tmp/ultrachat/",
        "prompt": "Once upon a time,",
        "max_tokens": 7,
        "temperature": 0
    }'

{"id":"cmpl-08d5765ec6a84a58aecb42f72037881b","object":"text_completion","created":1759280481,"model":"/tmp/ultrachat/","choices":[{"index":0,"text":" there was a man who lived in","logprobs":null,"finish_reason":"length","stop_reason":null,"token_ids":null,"prompt_logprobs":null,"prompt_token_ids":null}],"service_tier":null,"system_fingerprint":null,"usage":{"prompt_tokens":5,"total_tokens":12,"completion_tokens":7,"prompt_tokens_details":null},"kv_transfer_params":null}
```

And it produces sensible output. Not quite yet ready to document yet though.
